TypeCobol/AntlrGrammar/Cobol2002CodeElements.g4

```diff
@@ -44,6 +45,13 @@ initializeStatement:
 	(THEN? REPLACING initializeReplacingDirective+)?
 	(THEN? TO? DEFAULT)?;
 
+// New Cobol v6.1 ALLOCATE statement to obtain dynamic storage.
+// 'INITIALIZED' is defined here as a contextual keyword. The dataNameReference is therefore not allowed to be named 'INITIALIZED' in this statement.
+allocateStatement:
+	ALLOCATE ((arithmeticExpression CHARACTERS) | { !string.Equals(CurrentToken.Text, "INITIALIZED", System.StringComparison.InvariantCultureIgnoreCase) }? dataNameReference)
```

Comment:
Use storageArea2 instead of dataNameReference.
storageArea2 is mapped to ReceivingStoreageArea.

Then use CoboLExpressionsBuilder.CreateStorageArea to instantiate ReceivingStorageArea.